docs/developer-guide/views-examples.md

```diff
@@ -0,0 +1,164 @@
+
+### Creating a plugin
+
+Before reading further, it may be useful to refresh Lua basics by reading the [Lua manual][learning-lua].
+
+### Adding configuration settings
+
+WIP: write some setting option examples
+
+WIP: explain why we merge the tables
+
+```lua
+local toolbar = common.merge({
+  -- plugin configuration options go here
+}, config.plugins.toolbar)
+```
+
+### Checking if the plugin directory exists
+
+The code below is used to check if every folder detected by `get_file_info()` 
+actually exists and if it doesn't, a `nil` value is returned.
+
+```lua
+local function get_plugin_directory()
+  local paths = {
+    USERDIR .. PATHSEP .. "plugins" .. PATHSEP .. "toolbar",
+    DATADIR .. PATHSEP .. "plugins" .. PATHSEP .. "toolbar"
+  }
+  for i, v in ipairs(paths) do
+    if system.get_file_info(v) then
+      return v
+    end
+  end
+  return nil
+end
+```
+
+### Creating the view
+
+The code below creates an instance of our custom Toolbar View, inherits the ToolbarView super-constructor, 
+specifies the desired icon font and assigns the icons to their respective commands.
+
+```lua
+local ToolBar = ToolbarView:extend()
+
+function ToolBar:new()
+  ToolBar.super.new(self)
+  self.toolbar_font = renderer.font.load(get_plugin_directory() .. PATHSEP .. "toolbar.ttf", style.icon_big_font:get_size())
+  self.toolbar_commands = {
+    {symbol = "", command = "core:open-log"},
+    {symbol = "", command = "core:new-doc"},
+    {symbol = "", command = ""},
+    {symbol = "", command = ""},
+    {symbol = "", command = ""},
+    {symbol = "", command = ""},
+  }
+end
+```
+
+`local ToolBar = ToolbarView:extend()` is used to create a new custom View and make it 
+inherit ToolbarView.
+
+The `renderer.font.load` function assigns the icon font to the View.
+
+The `get_plugin_directory()` function gets the plugin folder path.
+
+### Adding the view
+
+The code below saves the toolbar in an exported table, to allow any plugin to require it, thus accessing the bar.
+
+```lua
+toolbar.example_toolbar_view = ToolBar()
+```
+
+The code below splits the `TreeView` according to the `up` direction, adds the `toolbar.example_toolbar_view` and fixes its size along the Y-axis.
+
+```lua
+toolbar.example_toolbar_node = TreeView.node.b:split("up", toolbar.example_toolbar_view, {y = true})
+```
+
+`return toolbar` returns the toolbar table to whatever function called it.
+
+### Creating a custom icon font
+
+To prepare a simple icon font with Fontello, you can do the following:
+
+1. Go to [Fontello](https://fontello.com/)
+2. Drag and drop the desired icons in the light-gray rectangle in the middle of the page: ![Drag & Drop][drag-n-drop]
+
+3. Select the icons you want to include in the font
+4. Press on the `Customize Codes` tab on the top-left of the page and copy each icon glyph and 
+paste it into the corresponding place in the `toolbar_commands` table
+
+    ![Customize Codes][customize-codes]
```

Comment:
This screenshot is marginally better than nothing. Please put a screenshot of the entire web page and highlight the tab with a red box or something.